pagefile.go

```diff
@@ -0,0 +1,374 @@
+//
+// This file provides low level routines for handling incremental backup
+// Incremental file format is:
+// 4 bytes header with designation information, format version and magic number
+// 8 bytes uint file size
+// 4 bytes uint changed pages count N
+// (N * 4) bytes for Block Numbers of changed pages
+// (N * BlockSize) bytes for changed page data
+//
+
+package walg
+
+/*
+// This block is CGo Postgres page header parsing. This is comment in comment. Cool.
```

Comment:
Once again, I find myself doubtful of the utility of Cgo here given the simplicity of the code: https://github.com/golang/go/issues/16051, https://www.cockroachlabs.com/blog/the-cost-and-complexity-of-cgo/. Especially for a loop as inner as one-per-page, and given that it composes poorly with several other Go features.

Reply:
OK I'll rewrite CGo code to manual parsing by binary.LittleEndian.Uint32
Anyway we have this bit-twiddling even in C
result.lsn = (((uint64_t)data->pd_lsn_h) << 32) + ((uint64_t)data->pd_lsn_l);
so 9 Go calls to parse page header will not be significantly worse. 

These two lines https://github.com/secwall/barman/blob/master/bin/barman-incr#L168 is equivalent to CGo code and seem prettier.